src/test/java/com/reservation/studio/domain/StudioTest.java

```diff
@@ -0,0 +1,192 @@
+package com.reservation.studio.domain;
+
+import com.reservation.studio.domain.exception.ContactDetailsCannotBeBlankException;
+import com.reservation.studio.domain.exception.InvalidCancellationThresholdException;
+import com.reservation.studio.domain.exception.InvalidStudioNameException;
+import com.reservation.studio.domain.exception.StudioAlreadyActiveException;
+import com.reservation.studio.domain.exception.StudioAlreadyClosedException;
+import org.junit.jupiter.api.Test;
+
+import java.time.DayOfWeek;
+import java.time.Duration;
+import java.time.LocalTime;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.api.Assertions.assertAll;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+class StudioTest {
+    @Test
+    void shouldCreateStudioAndEmitCreatedEvent() {
+        Studio studio = new StudioTestBuilder().build();
+
+        assertAll(
+                () -> assertFalse(studio.isActive()),
+                () -> assertThat(studio.pullEvents()).anyMatch(e -> e instanceof StudioCreated)
```

Comment:
```suggestion
                () -> assertThat(studio.pullEvents()).anyMatch(StudioCreated.class::isInstance)
```
IMHO będzie ładniej 😁